controllers/databaseengine_controller.go

```diff
@@ -70,14 +82,20 @@ func (r *DatabaseEngineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			Namespace: req.NamespacedName.Namespace,
 		},
 	}
-	_, err := controllerutil.CreateOrUpdate(ctx, r.Client, dbEngine, func() error {
+	if _, err := controllerutil.CreateOrUpdate(ctx, r.Client, dbEngine, func() error {
 		dbEngine.Spec.Type = engineType
 		return nil
-	})
-	if err != nil {
+	}); err != nil {
 		return ctrl.Result{}, err
 	}
 
+	defer func() {
+		if updErr := r.Status().Update(ctx, dbEngine); updErr != nil {
+			res = ctrl.Result{}
+			err = updErr
+		}
+	}()
```

Comment:
Without this there were multiple calls to .Status().Update(), so just wanted to consolidate everything in a single place. But yeah, I get that this might make things confusing together with the named returns.

I've fixed it now